src/tools/storage-key-recipe-resolver.ts

```diff
@@ -148,8 +151,15 @@ export class StorageKeyRecipeResolver {
             `Handle ${handle.localName} mapped to ephemeral handle '${match.id}'.`
           );
         }
+
+        if (isVolatileCreateHandle(match)) {
```

Comment:
sorry for the late insight. could you check `match.storageKey instanceOf VolatileStorageKey`, instead of having a helper method looking at capabilities?

Reply:
I tried that in this PR before. Unfortunately, it's not guaranteed that `match` has a `storageKey` set. If it did, I would re-use the `isVolatileStore` implementation in Manifest.ts (with some minor modification).

Reply:
if it doesn't have a `storageKey` doesn't it mean that it's a handle from a non long-running arc? so it cannot be mapped, but for a different reason.

Reply:
I'll do my best to explain my reasoning here, it might be better to VC. 

It doesn't have a storage key at this time since we create the storage keys later in this module. First we validate (this step), then we resolve the recipe, then we create the keys for create handles. We do these steps in a loop for each recipe. 

Before this check, we'd get an error at the resolution step. Adding this to the validation offers cleaner error messaging. 

I tried a solution that was dependent on the storage keys being present, but it was complicated by the existing order. Changing the order of processing would be unraveling. I tried null-checking the keys, but then that makes the system dependent on the order of recipes, which is fragile.

Reply:
i see. 

this approach to checking whether handle is mappable or not is very limited (volatile is not the only reason resolution might fail) and hacky  (reasoning about volatile in general is not absolutely correct atm), I'd rather we don't do it.
if you remove this check, the test will fail with `cannot find associated store` (thrown in recipe-resolver). I think if you were to add the `id` of the store that couldn't be found, that would significantly improve the error message without compromising code quality. in general, it should be recipe-resolver's job to report a comprehensive error if resolution fails.

Reply:
I'm happy with going this direction. I'll get on it now.